R/florabank_queries.R

```diff
@@ -0,0 +1,424 @@
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' # connect to florabank
+#' db_connectie <- connect_inbo_dbase("D0021_00_userFlora")
+#'
+#' # get records at 1 km x 1 km resolution for vascular plants from 2010 (default)
+#' fb_kwartier <- florabank_taxon_ifbl_year(db_connectie)
+#'
+#' # get records at 4 km x 4 km resoltion starting from 2000
+#' fb_uur <- florabank_taxon_ifbl_year(db_connectie, begin_year = 2000,
+#'  ifbl_resolution = "4km-by-4km", taxongroup = "Mossen")
+#'
+#' # disconnect from florabank
+#' dbDisconnect(db_connectie)
+#' }
+
+florabank_taxon_ifbl_year <- function(connection,
+                                      begin_year = 2010,
+                                      ifbl_resolution = c("1km-by-1km",
+                                                          "4km-by-4km"),
+                                      taxongroup = c("Vaatplanten",
+                                                     "Mossen",
+                                                     "Lichenen (korstmossen)",
+                                                     "Kranswieren")) {
+
+  assert_that(class(connection)[1] == "Microsoft SQL Server",
+              msg = "Not a connection object to database.")
+  assert_that(connection@info$dbname == "D0021_00_userFlora")
+
+  assert_that(is.numeric(begin_year))
+  assert_that(begin_year <= as.numeric(format(Sys.Date(), '%Y')))
+
+  ifbl_resolution = match.arg(ifbl_resolution)
+  taxongroup = match.arg(taxongroup)
+
+
+
+  if (ifbl_resolution == "4km-by-4km") {
+    glue_statement <- glue_sql(
+      "SELECT DISTINCT
+             tblIFBLHok.Code AS hok
+    , Year(tblWaarneming.BeginDatum) AS Jaar
+    , relTaxonTaxon.TaxonIDParent
+    , tblTaxon.Code AS Taxoncode
+    FROM
+    (((tblMeting INNER JOIN
+    (tblIFBLHok INNER JOIN tblWaarneming ON tblIFBLHok.ID = tblWaarneming.IFBLHokID)
+    ON tblMeting.WaarnemingID = tblWaarneming.ID)
+    INNER JOIN relTaxonTaxon ON tblMeting.TaxonID = relTaxonTaxon.TaxonIDChild)
+    INNER JOIN tblTaxon ON relTaxonTaxon.TaxonIDParent = tblTaxon.ID)
+    INNER JOIN relTaxonTaxonGroep ON tblTaxon.ID = relTaxonTaxonGroep.TaxonID
+    INNER JOIN tblTaxonGroep ON relTaxonTaxonGroep.TaxonGroepID = tblTaxonGroep.ID
+    WHERE
+    tblIFBLHok.Code LIKE '%-%' AND
+    tblTaxon.Code NOT LIKE '%-sp' AND
+    Year([tblWaarneming].[BeginDatum]) >={begin_year} AND
+    (Year([tblWaarneming].[BeginDatum])=Year([tblWaarneming].[EindDatum])) AND
+    (tblTaxonGroep.Naam={taxongroup}) AND
+    (tblMeting.MetingStatusCode='GDGA' OR tblMeting.MetingStatusCode='GDGK')
+    ORDER BY
+    Year(tblWaarneming.BeginDatum) DESC;",
+      begin_year = begin_year,
+      taxongroup = taxongroup,
+      .con = connection)
+    glue_statement <- iconv(glue_statement, from =  "UTF-8", to = "latin1")
+    query_result <- dbGetQuery(connection, glue_statement)
+
+
+    query_result <- query_result %>%
+      as_tibble %>%
+      mutate(ifbl_4by4 = str_sub(.data$hok, start = 1, end = 5))
```

Comment:
You can do this in the SQL statement using [substring](https://www.w3schools.com/sql/func_sqlserver_substring.asp). This removes the need for the stringr package